pages/index.tsx

```diff
@@ -43,8 +46,12 @@ const Index: NextPage<Props> = (props) => {
 
 Index.getInitialProps = async ({ req }) => {
   try {
+    const cookies = new Cookies(req?.headers.cookie);
+    const token = cookies.get(AUTH_COOKIE_NAME);
+    const payload = jwt.decode(token) as jwt.JwtPayload;
+
     const axios = createAxiosInstance(req);
-    await axios.get('/api/users/');
+    await axios.get('/api/users/' + payload.user_id);
```

Comment:
How does this work?

In my local, my JWT is: `eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJ1c2VyX2lkIjoxLCJpYXQiOjE2Nzk3NzE0NjUsImV4cCI6MTY4MDM3NjI2NX0.aXXWWhweVc2R_mC8KtY9hqIGP8HlQQN4Ug5vv9OhqP4`

Which is decoded to

```JS

{
  "user_id": 1,
  "iat": 1679771465,
  "exp": 1680376265
}

```
How does whatever is issuing the JWT know my user_id?

Reply:
The token is created when you login and it is sent to you as a cookie. Subsequent requests will include that cookie, thus letting the server know who you are.

During login (`/api/auth/login`), a `User` is found or created from your Discord user id:
https://github.com/timmyichen/dev-directory/blob/main/server/api/auth/auth.controller.ts#L37
https://github.com/timmyichen/dev-directory/blob/main/server/lib/auth.ts#L51-L59

That `User` (which includes your local user `id`) is then passed to a function that generates the JWT token:
https://github.com/timmyichen/dev-directory/blob/main/server/api/auth/auth.controller.ts#L38
https://github.com/timmyichen/dev-directory/blob/main/server/lib/auth.ts#L14-L18

Finally, that JWT token is then set as a cookie in the response:
https://github.com/timmyichen/dev-directory/blob/main/server/api/auth/auth.controller.ts#L40